perfmetrics/scripts/testing_on_gke/examples/fio/bq_utils.py

```diff
@@ -199,6 +202,71 @@ def _setup_dataset_and_tables(self):
       print(f'Failed to create fio table {self.table_id}: {e}')
       raise
 
+  def _delete_rows_incomplete_transaction(self, experiment_id=None):
+    """Helper function for _insert_row.
+
+    If insertion of some nth row fails, this method deletes (n-1) rows that were
+    inserted before
+
+    Args:
+      experiment_id (str): experiment_id of the experiment for which results are
+        being uploaded
+    """
+    if config_id:
+      query_delete_if_row_exists = """
+        DELETE FROM `{}.{}.{}`
+        WHERE experiment_id = '{}'
+      """.format(self.project_id, self.dataset_id, self.table_id, experiment_id)
+      job = self._execute_query(query_delete_if_row_exists)
+
+  def _insert_rows(self, rows_to_insert, experiment_id=None):
+    """Insert rows in table.
+
+    If insertion of some nth row fails, delete (n-1) rows that were inserted
+    before and raise an exception
+
+    Args:
+      rows_to_insert (str): Rows to insert in the table
+      experiment_id (str): experiment_id of the experiment for which results are
+        being uploaded
+
+    Raises:
+      Exception: If some row insertion failed.
+    """
+    table = self._get_table_from_table_id(self.table_id)
+    try:
+      result = self.client.insert_rows(table, rows_to_insert)
+      if result:
+        raise Exception(f'{result}')
+    except Exception as e:
+      self._delete_rows_incomplete_transaction(self.table_id, experiment_id)
+      raise Exception(f'Error inserting data to BigQuery table: {e}')
```

Comment:
Add table name here.

```suggestion
      raise Exception(f'Error inserting data to BigQuery table {self.project-id}:{self.dataset_id}{self.table_id}: {e}')
```